src/pytest_cov/plugin.py

```diff
@@ -234,7 +235,10 @@ def pytest_runtestloop(self, session):
                 message = 'Failed to generate report: %s\n' % exc
                 session.config.pluginmanager.getplugin("terminalreporter").write(
                     'WARNING: %s\n' % message, red=True, bold=True)
-                session.config.warn(code='COV-2', message=message)
+                if pytest.__version__ >= '3.8':
+                    warnings.warn(pytest.PytestWarning(message))
+                else:
+                    session.config.warn(code='COV-2', message=message)
```

Comment:
Might be good to have this pattern in the deprecation docs.

Reply:
Good idea, done: https://github.com/pytest-dev/pytest/pull/4029